### ISSUE TITLE
It seems at least the current Lineage 17.1 needs HTTPS

### ISSUE THREAD

**Issue by simonmicro** [julianxhokaxhiu/LineageOTA#65]
Well, when I used HTTP only (with an invalid SSL cert) the updater showed an "problem" and could not perform the update query. Changing the url inside the props to HTTPS and providing an valid SSL cert fixed that. I guess that should be noted in the `README.md`...

**Comment by kangear** [comment]
Had same problem.

**Comment by julianxhokaxhiu** [comment]
HTTPS should be provided through a reverse proxy :) See https://github.com/julianxhokaxhiu/vps-powered-by-docker and https://github.com/julianxhokaxhiu/vps-powered-by-docker/blob/master/modules/lineage.lan.sh as an example

**Comment by kangear** [comment]
```
https://download.lineageos.org/api/v1/shamu/nightly/77433d9634
```
```json
{
response: [
{
datetime: 1603238400,
filename: "lineage-17.1-20201021-nightly-shamu-signed.zip",
id: "24dfa87659f4fac3823bb9f4956708221a12bedc2cdf4db07e3b100f97adb8bf",
romtype: "nightly",
size: 453641780,
url: "https://mirrorbits.lineageos.org/full/shamu/20201021/lineage-17.1-20201021-nightly-shamu-signed.zip",
version: "17.1"
},
{
datetime: 1603843200,
filename: "lineage-17.1-20201028-nightly-shamu-signed.zip",
id: "ec7fa2faceb678aa14f423b790869046aecb11b13ad67196d11798521102da76",
romtype: "nightly",
size: 454158841,
url: "https://mirrorbits.lineageos.org/full/shamu/20201028/lineage-17.1-20201028-nightly-shamu-signed.zip",
version: "17.1"
},
{
datetime: 1604448000,
filename: "lineage-17.1-20201104-nightly-shamu-signed.zip",
id: "3aec875a6e9a9da72b6637db1963e028b0463e1d7b35ce31202ef0840f2d978c",
romtype: "nightly",
size: 455662283,
url: "https://mirrorbits.lineageos.org/full/shamu/20201104/lineage-17.1-20201104-nightly-shamu-signed.zip",
version: "17.1"
},
{
datetime: 1605052800,
filename: "lineage-17.1-20201111-nightly-shamu-signed.zip",
id: "1e1ee39a21bb15c1866b18b85a4e973dca145c12e4ea6cbfcd96ea3ce0ae102b",
romtype: "nightly",
size: 455814163,
url: "https://mirrorbits.lineageos.org/full/shamu/20201111/lineage-17.1-20201111-nightly-shamu-signed.zip",
version: "17.1"
}
]
}
```
like above example.
response[0].url need httpS. but it was http. even api url is httpS.

**Comment by simonmicro** [comment]
> HTTPS should be provided through a reverse proxy :) See https://github.com/julianxhokaxhiu/vps-powered-by-docker and https://github.com/julianxhokaxhiu/vps-powered-by-docker/blob/master/modules/lineage.lan.sh as an example

Well, even your docker container uses Apache under the hood! Therefore I just uploaded your api script to my shared hoster (for better reliability) and created my subdomain. For testing I, at first, disabled SSL to analyze the requests (just personal interest), but this caused the requests to never show up, because the Lineage Updater blocks unsecured connections (disabled inside the `config.xml` of the apk package).

What would very helpful, is a little note (at How to use/Composer) inside the `README.md` that using HTTPS is **REQUIRED** (and therefore a reverse proxy for the docker version is needed). Also you really should update the `system.prop` example accordingly (Idk what CyanogenMod thinks about the HTTPS enforcement) as it is currently broken!

```ini
# ...
lineage.updater.uri=http://my.ota.uri/api/v1/{device}/{type}/{incr}
# ...
```
->
```ini
# ...
lineage.updater.uri=https://my.ota.uri/api/v1/{device}/{type}/{incr}
# ...
```

I hope this helps to clarify what I wanted with this issue - otherwise great working api endpoint!

**Comment by julianxhokaxhiu** [comment]
Thank you @Simonmicro appreciated this project is useful to you :)

Would you mind opening a PR? I'd be really happy to review it.

Thank you in advance in case.

**Comment by julianxhokaxhiu** [comment]
Solved by #66
